domain/src/main/java/camp/nextstep/edu/domain/calculator/ExpressionParser.kt

```diff
@@ -0,0 +1,40 @@
+package camp.nextstep.edu.domain.calculator
+
+
+class ExpressionParser {
+
+    fun parse(
+        expression: String
+    ): List<ExpressionItem> {
+        require(expression.isNotEmpty()) { "공백은 입력할 수 없습니다." }
+
+        val items = expression.trim().split(" ")
+        require(isValidExpression(items)) { "유효하지 않은 표현식입니다." }
+
+        return toExpressionItem(items)
+    }
+
+    private fun toExpressionItem(
+        items: List<String>
+    ): List<ExpressionItem> {
+        val temp = mutableListOf<ExpressionItem>()
+        items.forEach {
+            temp.add(
+                if (it.isNumeric())
+                    Num(it.toInt())  // Numbers
+                else
+                    Operators.of(it) // Operators
```

Comment:
```suggestion
                if (it.isNumeric()) {
                    Num(it.toInt())
                } else {
                    Operators.of(it)
                }
```

라인을 바꿔주는 경우에는 괄호를 묶어주는 것이 가독성에 조금 더 좋은 것 같다는 생각이 듭니다!
각 조건 안에 들어갈 내용이 싱글 라인인 경우 
`if (it.isNumberic()) Num(it.toInt())` 
`else Operators.of(it)` 
같은 형식도 개인적으로는 애용하고 있습니다.
이 부분은 취향적인 부분이라서 반영하지 않고 참고만 해주셔도 좋을 것 같습니다!

추가로 객체 타입이 정확하게 명시되어있어서 주석은 달지 않아도 분명하게 코드를 이해할 수 있을 것 같아요!